apps/server/src/modules/tool/external-tool/uc/external-tool.uc.spec.ts

```diff
@@ -380,6 +443,88 @@ describe('ExternalToolUc', () => {
 
 			expect(result).toEqual(updatedExternalToolDO);
 		});
+
+		describe('when tool has no logo url', () => {
+			const setup2 = () => {
+				const user: User = userFactory.buildWithId();
+				const currentUser: ICurrentUser = { userId: user.id } as ICurrentUser;
+
+				const existingExternalTool: ExternalTool = externalToolFactory.buildWithId();
+				const existingExternalToolId = existingExternalTool.id as string;
+				const externalToolToUpdate: ExternalToolUpdate = {
+					...existingExternalTool,
+					id: existingExternalToolId,
+					logoUrl: undefined,
+				};
+
+				authorizationService.getUserWithPermissions.mockResolvedValue(user);
+				externalToolService.findExternalToolById.mockResolvedValueOnce(existingExternalTool);
+
+				return {
+					currentUser,
+					existingExternalToolId,
+					externalToolToUpdate,
+				};
+			};
+
+			it('should not fetch the logo', async () => {
+				const { currentUser, externalToolToUpdate, existingExternalToolId } = setup2();
+
+				await uc.updateExternalTool(currentUser.userId, existingExternalToolId, externalToolToUpdate);
+
+				expect(externalToolService.fetchBase64Logo).not.toHaveBeenCalled();
+			});
+		});
+
+		describe('when tool has a logo url', () => {
+			const setup2 = () => {
```

Comment:
setup2 sounds bad. rename to something more fitting, maybe "setupLogo" like further down?